packages/immutadot-parser/src/parser.utils.ts

```diff
@@ -0,0 +1,95 @@
+import { NavType } from "./enums";
+
+const mapIfNotNull = <T, R> (maybe: T, fn: (v: T) => R): R => maybe === null ? null : fn(maybe);
+
+type Parser = (str: string) => string[];
```

Comment:
Some `Parser`s return `null` so return type should be `string[] | null`. Also I think it would useful to alias this return type to `ParseResult` or some other name.

Reply:
Make it generic! :)
```
type Parser<T> = (input: string) => T | null
```
Follow by making all of the function working on parsers (filter, map, ...) generic too. All of the code in this file should not be concerned about paths.

Reply:
I agree utils should not be concerned with paths...